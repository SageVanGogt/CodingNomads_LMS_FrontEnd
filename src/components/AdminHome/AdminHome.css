.Teacher_home-container {
  width: calc(100vw - 200px);
  height: 100vh;
  margin-left: 200px;
  background-color: #efefef;
  padding-top: 40px; }

.Teacher_board {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  margin: auto;
  width: 70vw;
  height: 80vh;
  background-color: #efefef; }

.Teacher_view {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 50px;
  width: 400px;
  height: 250px;
  color: white;
  background-color: grey;
  font-family: 'Open Sans', sans-serif; }

.Teacher_view h3 {
  font-size: 2rem; }

.Teacher_options {
  visibility: hidden;
  position: absolute;
  z-index: 1;
  width: 150px;
  margin-left: 250px;
  height: 250px;
  background-color: #0F94FB50; }

.Teacher_options div {
  height: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer; }

.Teacher_view:hover .Teacher_options {
  visibility: visible; }

.Teacher_options div:hover, img:hover {
  background-color: #0F94FB; }

.add-btn {
  font-size: 3rem; }
