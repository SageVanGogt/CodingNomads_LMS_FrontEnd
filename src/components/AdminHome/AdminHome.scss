.Teacher_board {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  width: 80vw;
  margin-left: 200px;
}

.Teacher_view {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 50px;
  width: 400px;
  height: 250px;
  background-color: grey;
}

.Teacher_options {
  visibility: hidden;
  position: fixed;
  z-index: 1;
  width: 150px;
  margin-left: 250px;
  height: 250px;
  background-color: #0F94FB50;
}

.Teacher_options div {
  height: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.Teacher_view:hover .Teacher_options {
  visibility: visible
}

.Teacher_options div:hover {
  background-color: #0F94FB;
}