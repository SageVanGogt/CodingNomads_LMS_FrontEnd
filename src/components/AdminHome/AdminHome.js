import React, { Component} from 'react';
import { connect } from 'react-redux';
import './AdminHome.css';
import { Link } from 'react-router-dom';

export class AdminHome extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div className="Teacher_home">
        <section className="Teacher_board">
          <article className="Teacher_view">
            <h3>courses</h3>
            <div className="Teacher_options">
              <div>add</div>
              <div>view</div>
            </div>
          </article>
          <article className="Teacher_view">
            <h3>notifications</h3>
            <div className="Teacher_options">
              <div>add</div>
              <div>view</div>
            </div>
          </article>
          <article className="Teacher_view">
            <h3>tasks</h3>
            <div className="Teacher_options">
              <div>add</div>
              <div>view</div>
            </div>
          </article>
          <article className="Teacher_view">
            <h3>students</h3>
            <div className="Teacher_options">
              <div>add</div>
              <div>view</div>
            </div>
          </article>
        </section>
      </div>
    );
  }
}

export default AdminHome;