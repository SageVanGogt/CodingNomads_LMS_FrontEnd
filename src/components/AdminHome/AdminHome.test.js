import React from 'react';
import { shallow } from 'enzyme';
import { AdminHome, mapDispatchToProps } from './AdminHome';

describe('AdminHome', () => {
  let wrapper;
  let mockHistory;
  let mockEvent;

  beforeEach(() => {
    mockEvent = {
      target: {
        id: 'COURSES'
      }
    };
    mockHistory = {
      push: jest.fn()
    };
    wrapper = shallow(<AdminHome event={mockEvent} history={mockHistory}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
  describe('handleRoute', () => {
    it('should call push on history with the correct params, to redirect the page', () => {
      let expected = '/courses';
      wrapper.instance().handleRoute(mockEvent);

      expect(mockHistory.push).toHaveBeenCalledWith(expected);
    })
  })
  
});