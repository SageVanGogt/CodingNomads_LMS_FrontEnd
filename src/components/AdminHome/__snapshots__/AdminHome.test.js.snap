// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHome should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminHome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="Teacher_board"
>
        <article
                className="Teacher_view"
        >
                <h3>
                        courses
                </h3>
                <div
                        className="Teacher_options"
                >
                        <div>
                                add
                        </div>
                        <div>
                                view
                        </div>
                </div>
        </article>
        <article
                className="Teacher_view"
        >
                <h3>
                        notifications
                </h3>
                <div
                        className="Teacher_options"
                >
                        <div>
                                add
                        </div>
                        <div>
                                view
                        </div>
                </div>
        </article>
        <article
                className="Teacher_view"
        >
                <h3>
                        tasks
                </h3>
                <div
                        className="Teacher_options"
                >
                        <div>
                                add
                        </div>
                        <div>
                                view
                        </div>
                </div>
        </article>
        <article
                className="Teacher_view"
        >
                <h3>
                        students
                </h3>
                <div
                        className="Teacher_options"
                >
                        <div>
                                add
                        </div>
                        <div>
                                view
                        </div>
                </div>
        </article>
</section>,
      "className": "Teacher_home",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="Teacher_view"
>
            <h3>
                        courses
            </h3>
            <div
                        className="Teacher_options"
            >
                        <div>
                                    add
                        </div>
                        <div>
                                    view
                        </div>
            </div>
</article>,
          <article
            className="Teacher_view"
>
            <h3>
                        notifications
            </h3>
            <div
                        className="Teacher_options"
            >
                        <div>
                                    add
                        </div>
                        <div>
                                    view
                        </div>
            </div>
</article>,
          <article
            className="Teacher_view"
>
            <h3>
                        tasks
            </h3>
            <div
                        className="Teacher_options"
            >
                        <div>
                                    add
                        </div>
                        <div>
                                    view
                        </div>
            </div>
</article>,
          <article
            className="Teacher_view"
>
            <h3>
                        students
            </h3>
            <div
                        className="Teacher_options"
            >
                        <div>
                                    add
                        </div>
                        <div>
                                    view
                        </div>
            </div>
</article>,
        ],
        "className": "Teacher_board",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                courses
</h3>,
              <div
                className="Teacher_options"
>
                <div>
                                add
                </div>
                <div>
                                view
                </div>
</div>,
            ],
            "className": "Teacher_view",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "courses",
              },
              "ref": null,
              "rendered": "courses",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    add
</div>,
                  <div>
                    view
</div>,
                ],
                "className": "Teacher_options",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "add",
                  },
                  "ref": null,
                  "rendered": "add",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "view",
                  },
                  "ref": null,
                  "rendered": "view",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                notifications
</h3>,
              <div
                className="Teacher_options"
>
                <div>
                                add
                </div>
                <div>
                                view
                </div>
</div>,
            ],
            "className": "Teacher_view",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "notifications",
              },
              "ref": null,
              "rendered": "notifications",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    add
</div>,
                  <div>
                    view
</div>,
                ],
                "className": "Teacher_options",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "add",
                  },
                  "ref": null,
                  "rendered": "add",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "view",
                  },
                  "ref": null,
                  "rendered": "view",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                tasks
</h3>,
              <div
                className="Teacher_options"
>
                <div>
                                add
                </div>
                <div>
                                view
                </div>
</div>,
            ],
            "className": "Teacher_view",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "tasks",
              },
              "ref": null,
              "rendered": "tasks",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    add
</div>,
                  <div>
                    view
</div>,
                ],
                "className": "Teacher_options",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "add",
                  },
                  "ref": null,
                  "rendered": "add",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "view",
                  },
                  "ref": null,
                  "rendered": "view",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                students
</h3>,
              <div
                className="Teacher_options"
>
                <div>
                                add
                </div>
                <div>
                                view
                </div>
</div>,
            ],
            "className": "Teacher_view",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "students",
              },
              "ref": null,
              "rendered": "students",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    add
</div>,
                  <div>
                    view
</div>,
                ],
                "className": "Teacher_options",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "add",
                  },
                  "ref": null,
                  "rendered": "add",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "view",
                  },
                  "ref": null,
                  "rendered": "view",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="Teacher_board"
>
          <article
                    className="Teacher_view"
          >
                    <h3>
                              courses
                    </h3>
                    <div
                              className="Teacher_options"
                    >
                              <div>
                                        add
                              </div>
                              <div>
                                        view
                              </div>
                    </div>
          </article>
          <article
                    className="Teacher_view"
          >
                    <h3>
                              notifications
                    </h3>
                    <div
                              className="Teacher_options"
                    >
                              <div>
                                        add
                              </div>
                              <div>
                                        view
                              </div>
                    </div>
          </article>
          <article
                    className="Teacher_view"
          >
                    <h3>
                              tasks
                    </h3>
                    <div
                              className="Teacher_options"
                    >
                              <div>
                                        add
                              </div>
                              <div>
                                        view
                              </div>
                    </div>
          </article>
          <article
                    className="Teacher_view"
          >
                    <h3>
                              students
                    </h3>
                    <div
                              className="Teacher_options"
                    >
                              <div>
                                        add
                              </div>
                              <div>
                                        view
                              </div>
                    </div>
          </article>
</section>,
        "className": "Teacher_home",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="Teacher_view"
>
              <h3>
                            courses
              </h3>
              <div
                            className="Teacher_options"
              >
                            <div>
                                          add
                            </div>
                            <div>
                                          view
                            </div>
              </div>
</article>,
            <article
              className="Teacher_view"
>
              <h3>
                            notifications
              </h3>
              <div
                            className="Teacher_options"
              >
                            <div>
                                          add
                            </div>
                            <div>
                                          view
                            </div>
              </div>
</article>,
            <article
              className="Teacher_view"
>
              <h3>
                            tasks
              </h3>
              <div
                            className="Teacher_options"
              >
                            <div>
                                          add
                            </div>
                            <div>
                                          view
                            </div>
              </div>
</article>,
            <article
              className="Teacher_view"
>
              <h3>
                            students
              </h3>
              <div
                            className="Teacher_options"
              >
                            <div>
                                          add
                            </div>
                            <div>
                                          view
                            </div>
              </div>
</article>,
          ],
          "className": "Teacher_board",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  courses
</h3>,
                <div
                  className="Teacher_options"
>
                  <div>
                                    add
                  </div>
                  <div>
                                    view
                  </div>
</div>,
              ],
              "className": "Teacher_view",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "courses",
                },
                "ref": null,
                "rendered": "courses",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      add
</div>,
                    <div>
                      view
</div>,
                  ],
                  "className": "Teacher_options",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "add",
                    },
                    "ref": null,
                    "rendered": "add",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "view",
                    },
                    "ref": null,
                    "rendered": "view",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  notifications
</h3>,
                <div
                  className="Teacher_options"
>
                  <div>
                                    add
                  </div>
                  <div>
                                    view
                  </div>
</div>,
              ],
              "className": "Teacher_view",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "notifications",
                },
                "ref": null,
                "rendered": "notifications",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      add
</div>,
                    <div>
                      view
</div>,
                  ],
                  "className": "Teacher_options",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "add",
                    },
                    "ref": null,
                    "rendered": "add",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "view",
                    },
                    "ref": null,
                    "rendered": "view",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  tasks
</h3>,
                <div
                  className="Teacher_options"
>
                  <div>
                                    add
                  </div>
                  <div>
                                    view
                  </div>
</div>,
              ],
              "className": "Teacher_view",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "tasks",
                },
                "ref": null,
                "rendered": "tasks",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      add
</div>,
                    <div>
                      view
</div>,
                  ],
                  "className": "Teacher_options",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "add",
                    },
                    "ref": null,
                    "rendered": "add",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "view",
                    },
                    "ref": null,
                    "rendered": "view",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  students
</h3>,
                <div
                  className="Teacher_options"
>
                  <div>
                                    add
                  </div>
                  <div>
                                    view
                  </div>
</div>,
              ],
              "className": "Teacher_view",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "students",
                },
                "ref": null,
                "rendered": "students",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      add
</div>,
                    <div>
                      view
</div>,
                  ],
                  "className": "Teacher_options",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "add",
                    },
                    "ref": null,
                    "rendered": "add",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "view",
                    },
                    "ref": null,
                    "rendered": "view",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "article",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
